settings/post-process-display.sql

```diff
@@ -0,0 +1,249 @@
+    coalesce(i_stops.name, i_positions.member_type::text || '_' || i_positions.member_osm_id::text),
+    ST_DistanceSphere(
+      ST_Transform(i_positions.geom, 4326),
+      ST_Transform(ST_LineInterpolatePoint(i_ways.geom, ST_LineLocatePoint(i_ways.geom, i_positions.geom)), 4326)
+    )
+  ) AS t
+GROUP BY
+  geom
+;
+CREATE INDEX idx_d_routes_position_geom ON d_routes_position USING GIST(geom);
+DROP SEQUENCE IF EXISTS d_routes_position_id_seq;
+CREATE SEQUENCE d_routes_position_id_seq;
+ALTER TABLE d_routes_position ADD COLUMN id integer NOT NULL DEFAULT nextval('d_routes_position_id_seq');
+
+-- Compute route stop positions list for each route
+DROP TABLE IF EXISTS d_routes_position_ids;
+CREATE TABLE d_routes_position_ids AS
+SELECT
+  rel_osm_id,
+  array_agg(id) AS positions_ids
+FROM (
+  SELECT
+    id,
+    unnest(rels_osm_id) AS rel_osm_id
+  FROM
+    d_routes_position
+) AS t
+GROUP BY
+  rel_osm_id
+;
+ALTER TABLE d_routes_position DROP COLUMN rels_osm_id;
+
+
+-- Collect and complete stops
+DROP TABLE IF EXISTS d_stops;
+CREATE TABLE d_stops AS
+SELECT
+  *,
+  NULL::int AS max_diameter,
+  NULL::int AS max_avg_distance,
+  NULL::int AS number_of_routes,
+  NULL::text[][] AS routes_ref_colour
+FROM
+  i_stops
+;
+CREATE INDEX idx_d_stops_geom ON d_stops USING GIST(geom);
+
+
+DROP TABLE IF EXISTS t_stops_routes;
+CREATE TEMP TABLE t_stops_routes AS
+SELECT
+  d_stops.osm_type,
+  d_stops.osm_id,
+  max(i_routes.diameter) AS max_diameter,
+  max(i_routes.avg_distance) AS max_avg_distance,
+  count(*) AS number_of_routes,
+  array_agg(DISTINCT array[i_routes.ref, i_routes.colour]) AS routes_ref_colour
+FROM
+  d_stops
+  JOIN i_positions ON
+    i_positions.member_type = d_stops.osm_type AND
+    i_positions.member_osm_id = d_stops.osm_id
+  JOIN i_routes ON
+    i_routes.osm_id = i_positions.rel_osm_id
+GROUP BY
+  d_stops.osm_type,
+  d_stops.osm_id
+;
+
+UPDATE
+  d_stops
+SET
+  max_diameter = dt.max_diameter,
+  max_avg_distance = dt.max_avg_distance,
+  number_of_routes = dt.number_of_routes,
+  routes_ref_colour = dt.routes_ref_colour
+FROM
+  t_stops_routes AS dt
+WHERE
+  dt.osm_type = d_stops.osm_type AND
+  dt.osm_id = d_stops.osm_id
+;
+
+DROP TABLE t_stops_routes;
+
+
+-- Compute stops shield
+DROP TABLE IF EXISTS d_stops_shield;
+CREATE TABLE d_stops_shield AS
+SELECT
+  osm_type,
+  osm_id,
+  i,
+  routes_ref_colour[i][1] AS ref,
+  CASE WHEN routes_ref_colour[i][2] IS NULL OR routes_ref_colour[i][2] = '' THEN '#4898ff' ELSE routes_ref_colour[i][2] END AS colour,
+  geom
+FROM (
+  SELECT
+    generate_subscripts(routes_ref_colour, 1) as i,
+    *
+  FROM
+    d_stops
+  ) AS t
+;
+ALTER TABLE d_stops DROP COLUMN routes_ref_colour;
+
+
+-- Hack to inject JSON into vtiles, need an API to remove this.
+DROP TABLE IF EXISTS t_stops_json;
+CREATE TEMP TABLE t_stops_json AS
+SELECT
+  json_agg(routes_at_stop) AS json,
+  osm_id
+FROM (
+  SELECT
+    row_to_json(t) AS routes_at_stop,
+    osm_id
+  FROM (
+    SELECT
```

Comment:
this `select` request is almost a duplication of the one used to create `t_stops_routes`. We should mutualize.
It would also allow to have the bus route in the same order in the pop-up and in the shields, and to have them deduplicated (they are duplicated in the pop-up).

Reply:
Part of the join is the same, not the rest.
* Shield and json content does not have particular order. We can order both.
* This part will be replaced by the API.

Reply:
Done swithed to gray.

Reply:
In fact, `ref` ordering is not so obvious. I suggest to stay as is for now. I add it to the todo list.